### ISSUE TITLE
downgrade python requirement from `>=3.12.1` to `~>=3.9.2`

### ISSUE THREAD

**Issue by el19yz** [el19yz/s5dev#1]
**tl;dr**
The package currently requires `python >= 3.12.1`, but the constraint should be able to be relaxed down to `~>=3.9.2`.

**Background**
Previously, this codebase was developed with `python=3.9.0`. However, upon upgrading core packages such as `flax` and `jax` (for cuda compatibility and to keep up with the latest stable releases), the training script would raise `TypeError`s.

These errors were traced back to a typing resolution error in `python=3.9.0`; this error was resolved in `python=3.9.2` (see: https://github.com/python/cpython/issues/87131)

In order to develop the features of the codebase reliant on GPU access, an HPC is used. This HPC provides shared software modules for easy development, but it only has python versions `3.9.0` and `3.12.1` available. In the interest of moving onto other higher priority features, the codebase was packaged up and only tested with `python=3.12.1`.

**Motivation**
It would be desirable to relax the python version requirement to be more reflective of the true requirements of the package, and make it more accessible or accepted by a broader user base.

**Suggested steps**
- [x] Setup virtual environment with `python==3.9.2` and install package and dependencies.
- [x] Verify non-destructive installation and by e.g. running the small (cpu) test
- [x] Upgrade `pyproject.toml`, `requirements.txt`, and other files as needed
